rust/agama-server/src/software/web.rs

```diff
@@ -213,6 +213,11 @@ pub async fn software_service(dbus: zbus::Connection) -> Result<Router, ServiceE
             "/registration",
             get(get_registration).post(register).delete(deregister),
         )
+        .route("/registration/addons", post(register_addon))
+        .route(
+            "/registration/addons/registered",
```

Comment:
I am not sure about this organization of URLs and HTTP verbs. At first glance, I would expect `POST /registration/addons` to create a new addon, and that's not the intention.

I think will need a way to retrieve the list of addons when we work on the UI, right? So I would reserve `/registration/addons` for that.

* `GET /registration/addons` for the list of addons (out of scope).
* `GET /registration/addons/registered` for the list of registered addons. If both endpoints will use the same format, you could use `GET /registration/addons?registered=true` instead. 
* `POST /registration/addons/register` (or perhaps a better naming) for the registration.

Does it make sense?

Reply:
I already thought about `/registration/addons?registered=true`, but the problem is that for the CLI we only need minimal data (name, version, reg. code), but for the full web UI we will need much more details, like description, if the extension is free or requires a registration code, license (optional), dependencies, whether it should be selected by default or not, if it is mirrored on the server (RMT only)... That's why it uses a different endpoint in the end.

About `POST /registration/addons/register`, yes, that makes more sense, I'll update the path.

Reply:
You can also have a query argument, which allows you to switch between a full representation and a summary. But we can discuss this later, when we implement the `/registration/addons` for the UI.